src/pfs/pfs.proto

```diff
@@ -300,6 +300,20 @@ message CreateBranchRequest {
   bool new_commit_set = 5; // overrides the default behavior of using the same CommitSet as 'head'
 }
 
+message SearchForFileInBranchRequest {
+  Commit start = 1;
+  string file_path = 2;
+  int32 limit = 3; // Does it make more sense to define a limit in terms of raw commits to search or in terms of the length of the returned 'found commits' array?
```

Comment:
Limit usually refers to the amount returned.  In SQL for example the user doesn't specify the maximum number of rows to search.

Maybe we should expose the maximum number to search as well, but the user probably cares more about a deadline for the rpc, than the amount of commits checked.  gRPC has the ability to send that over as metadata and we can get at it with the `context.Context` that is provided.

Reply:
Okay cool. Thanks for the input. I think based your comment, we'll leave the limit logic the way it is now. We can also pass in a timeout via the context instead of defining it as a parameter, that makes more sense.

Reply:
Our client wrapper isn't really idiomatic Go.  You can set the context, but not as an argument.  I've talked with @jrockway about plans to deprecate the client wrapper.  If we continue to support it, it needs to be reworked.  I wouldn't worry too much about this package/method.  Anyone seriously using the API will use gRPC directly, Hub (maybe the heaviest consumers of the APIs ever) is a good example.